src/test/java/com/desiato/puresynth/controllers/LoginControllerTest.java

```diff
@@ -0,0 +1,144 @@
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+public class LoginControllerTest extends BaseTest {
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    private static final Logger logger = LoggerFactory.getLogger(LoginControllerTest.class);
+
+
+    @Test
+    public void manuallyCreateAndPersistUser() {
+        User user = new User();
+        user.setEmail("example@example.com");
+        user.setPassword(passwordEncoder.encode("password123"));
+
+        userRepository.save(user);
+    }
+
+    @Test
+    public void authenticateUser_WhenPasswordIsInvalid_ShouldReturnUnauthorized() throws Exception {
+        String uniqueEmail = generateUniqueEmail();
+        String validPassword = "password123";
+        userService.createUser(uniqueEmail, passwordEncoder.encode(validPassword));
+
+        String invalidPassword = "invalidPassword";
+        String json = """
+                {
+                    "email": "%s",
+                    "password": "%s"
+                }
+                """.formatted(uniqueEmail, invalidPassword);
+
+        mockMvc.perform(post("/api/login")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(json))
+                .andExpect(status().isUnauthorized())
+                .andExpect(content().string("Unauthorized"));
+    }
+
+    @Test
+    public void authenticateUser_WhenCredentialsAreValid_ShouldAuthenticateSuccessfully() throws Exception {
+        String uniqueEmail = generateUniqueEmail();
+        String password = "password123";
+
+        logger.info("Created user with email: {}", uniqueEmail);
+
+        userService.createUser(uniqueEmail, passwordEncoder.encode(password));
+
+        String loginJson = String.format("""
+                {
+                    "email": "%s",
+                    "password": "%s"
+                }
+                """, uniqueEmail, password);
+
+        logger.info("Sending login request with email: {}", uniqueEmail);
+
+        MvcResult loginResult = mockMvc.perform(post("/api/login")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(loginJson))
+                .andExpect(status().isOk())
+                .andExpect(header().exists("Auth-Token"))
+                .andReturn();
+
+        logger.info("Login response status code: {}", loginResult.getResponse().getStatus());
+
+        String token = loginResult.getResponse().getHeader("Auth-Token");
+
+        assertNotNull(token, "Authentication token is missing or invalid");
```

Comment:
This should not be in a header, but in a response body.

Reply:
ok, modified